app/controllers/adventures_controller.rb

```diff
@@ -119,13 +122,17 @@ def adventure_params
       :theme,
       :password,
       :has_password,
+      :age_limit,
+      :has_age_limit,
       :show_source
     )
   end
 
   def check_authentication
     if !@adventure.editable_by?(current_user) && (@adventure.has_password? && !session[@adventure.title])
       render :authenticate_player
+    elsif !@adventure.editable_by?(current_user) && (@adventure.has_age_limit? && !session[@adventure.title])
```

Comment:
if an adventure has both a password and an age gate, I'm guessing, looking at this `check_authentication` method, that the user would only get hit with the password check. Can you split up the check for "has this user cleared this specific auth hurdle" to enable a story to have both checks in place?